lib/checks/keyboard/main-is-top-level.js

```diff
@@ -0,0 +1,13 @@
+var landmarks = axe.commons.aria.getRolesByType('landmark');
+var parent = node.parentElement;
+while (parent){
+	var role = parent.getAttribute('role');
+	if (!role){
+		role = axe.commons.aria.implicitRole(parent);
```

Comment:
WIlco,

1. In ARIA 1.1 application and presentation roles are not considered landmarks.
2. In the case of a form landmark:
2.a Form landmarks should be fairly rare, there is discussion in the ARIA and HTML5 working groups that form landmarks should/must have an accessible name to be considered a landmark or at least useful, since without knowing what the form is about the navigation feature is somewhat limited since getting to form controls is easy from screen readers and makes the landmark more of a distraction than a useful feature.
2.b Forms are typically just one part of a page so should be in another landmark, the main landmark is what people will be looking for, not a form landmark.  In general Form landmarks should only be used if there are multiple forms on a page.